### PR TITLE
5025 hidden items showing on item usage and stock details reports

### LINKED ISSUE #5025: Hidden items are appearing in item usage and stock details reports
## What went wrong? 😲

Hidden items are appearing in item usage and stock details reports. 
Screenshot 2024-10-07 at 10 09 02 PM

## Expected behaviour 🤔
If item is hidden it is probably not meant to be ordered etc. Would expect it to not be showing.

## How to Reproduce 🔨

Steps to reproduce the behaviour:

## Your environment 🌱

- Open mSupply Version:
- Legacy mSupply Central Server Version:

- Platform:

- Database type:

### PR DESCRIPTION
**Pull request by fergie-nz**
Fixes #5025

Removes hidden items from report rendering.

There might be some confusion about what visible means for items in oms. We don't sync the hidden boolean from OG. in OMS visible means being part of a masterlist which is visible on the store.

I've updated explanations in comments.

Also graphql report queries updated to only show items both active and with stock on hand.
I think this should be sufficient because items cannot be hidden on OG if there is stock available.

Also added default sorting ascending and by item name on all reports

# 👩🏻‍💻 What does this PR do?

## 💌 Any notes for the reviewer?

For reviewers - the fastest way to get the new report is like so:

- [ ] Do not stop your server running instance.
- [ ] Manually delete the item-usage report from your database
- [ ] Open a new terminal and run:

```
cargo build &&  ./target/debug/remote_server_cli build-standard-reports && ./target/debug/remote_server_cli upsert-reports-json
```

Which will rebuild and upsert reports to the database name set up in your .yaml file.

# 🧪 Testing

- [ ] Run commands above to delete, rebuild, and upsert reports
- [ ] Go to reports
- [ ] Render reports and don't see empty items

# 📃 Documentation

- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour

### CODE REVIEW

File: server/reports/stock-detail/2_3_0/src/query.graphql [hunk 1]
```diff
@@ -2,8 +2,11 @@ query StockDetail($storeId: String, $itemCodeOrName: String) {
   stockLines(
     storeId: $storeId
     page: { first: 5000 }
-    filter: { itemCodeOrName: { like: $itemCodeOrName }, hasPacksInStore: true }
-    sort: { key: itemName, desc: false }
+    filter: {
+      itemCodeOrName: { like: $itemCodeOrName }
+      hasPacksInStore: true
+      isActive: true
```
Comment (andreievg) [inline server/reports/stock-detail/2_3_0/src/query.graphql:8]: I got confused about this, i thought it won't work (graphql won't allow it), but it seems to work somehow ? (isActive is not part of filter input)

**Review by andreievg — COMMENTED** [review]
Looks great, one comment about isActive on stockLine, and do we need to commit standard_reports.json ?

File: server/reports/stock-detail/2_3_0/src/query.graphql (hunk 1 above)
Comment (roxy-dao) [reply server/reports/stock-detail/2_3_0/src/query.graphql:8]: Have added the filter on stock line

**Comment by andreievg** [comment]
I updated from 2.4.1 (to get the other report fix in) and then generated standard reports, thanks @fergie-nz thanks @roxy-dao

**Review by andreievg — APPROVED** [review]
Looks great thanks

**Pull request merged**